comit/src/network/orderbook.rs

```diff
@@ -102,21 +103,43 @@ impl Orderbook {
 
     /// Take an order, called by Alice i.e., the taker.
     /// Does _not_ remove the order from the order book.
-    pub fn take(&mut self, order_id: OrderId) -> anyhow::Result<()> {
+    pub fn take(
+        &mut self,
+        order_id: OrderId,
+        partial_take_amount: asset::Bitcoin,
```

Comment:
I think this parameter could be optional, in case you didn't want to take the order fully.

Reply:
Either we make the field on the wire also optional or we default to the full amount.